docs/tools/editor/vscode.md

```diff
@@ -1,79 +1,129 @@
 
-### 配置编译
+### Configuring Compilation
 
-首先用 VS Code 打开一个文件夹，然后按下<kbd>F1</kbd>，输入 `C/C++: Edit configurations (UI)`，进入 C/C++ 插件的设置界面。
+First, open a folder in VS Code, and then press <kbd>F1</kbd>, type `C/C++: Edit configurations (UI)` and enter the configuration page of C/C++ extension.
 
 ![vscode-3](images/vscode-3.png)
 
-在“编译器路径”中选择 G++ 或 Clang 的所在路径。如果没有可选项，请检查编译器所在路径是否添加到了操作系统的 `PATH` 变量中。
+Choose the path where G++ or Clang located in Compiler Path. re-check whether the path of compiler has been added to `PATH` environment variable of operating system if there is nothing to choose.
 
-### 配置 IntelliSense
+### Configuring IntelliSense
 
-用于调整 VS Code 的智能补全。
+This section is talking about adjusting smart auto-completion of VS Code.
 
-如果你使用 Clang 编译器，在“IntelliSense 模式”中选择 `clang-x64` 而非默认的 `msvc-x64`；如果你使用 G++ 编译器，选择 `gcc-x64` 以使用自动补全等功能。否则会得到“IntelliSense 模式 msvc-x64 与编译器路径不兼容。”的错误。
+If you are using Clang as compiler, remember to choose `clang-x64` instead of default `msvc-x64`, or for G++ compiler users, choose `gcc-x64`, to enable features like auto-completion. Otherwise you will encounter an error "Intellisense mode msvc-x64 is incompatible with compiler path".
 
 ![](images/vscode-4.png)
 
-### 配置 GDB/LLDB 调试器
+### Configure GDB/LLDB Debuggers
 
-在 VS Code 中新建一份 C++ 代码文件，按照 C++ 语法写入一些内容（如 `int main(){}`），保存并按下<kbd>F5</kbd>，进入调试模式。
-如果出现了“选择环境”的提示，选择“C++ (GDB/LLDB)”。在“选择配置”中，G++ 用户选择 `g++.exe - 生成和调试活动文件`；Clang 用户选择 `clang++ - 生成和调试活动文件`。
+To enter debugging mode, create a new C++ file in VS Code and write something in C++ grammar, press <kbd>F5</kbd> after saving file. If a **choose environment** dialog pops up, select **C++ (GDB/LLDB)** and choose **g++ - Build and debug active file** for G++ users, or **clang++ - Build and debug active file** for Clang users.
 
 ???+ warning
-    配置名称并非固定，而是可以自定义的。不同的操作系统可能具有不同的配置名称。
+    The configuration name is customizable. It may differ in different operating systems.
 
-完成后，VS Code 将自动完成初始化操作并弹出一个 `launch.json` 配置文件。关闭它。
+Then, VS Code will finish initialization automatically and open a `launch.json` configuration file. Ignore and close it.
 
-至此，所有的配置流程已经完毕。再次按下<kbd>F5</kbd>即可看到软件下方的调试信息。
+At this point, all configuration processes have been completed. Press <kbd>F5</kbd> again and you will see debugging informations.
 
-若要在以后使用 VS Code 编译并调试代码，所有的源代码都需要保存至这个文件夹内。若要编译并调试其他文件夹中存放的代码，需要重新执行上述步骤（或将旧文件夹内的 `.vscode` 子文件夹复制到新文件夹内）。
+All files needed to be compiled and debugged are required to be saved in this folder in the future. If you need to compile and debug codes in other folders, you need to perform above steps again, or copy `.vscode` sub-folder in original folder to new folder.
 
-### 开始调试代码
+### Debugging Code
 
-使用 VS Code 打开一份代码，将鼠标悬停在行数左侧的空白区域，并单击出现的红点即可为该行代码设置断点。再次单击可取消设置断点。
+To set a breakpoint on a certain row, open a code in VS Code, hover the cursor over the blank space to the left of the row number, and click the appeared red point. Re-click it will cancel the breakpoint.
```

Comment:
```suggestion
To set a breakpoint on a certain line, open a code in VS Code, hover the cursor over the blank space to the left of the row number, and click the appeared red point. Re-click it will cancel the breakpoint.
```